.github/workflows/build.yml

```diff
@@ -30,4 +39,44 @@ jobs:
       - name: Build Leo
         id: build-leo
         run: |
-          npm run transform-tokens
\ No newline at end of file
+          npm run transform-tokens
+
+      - name: Configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          role-session-name: github-actions-pr-leo-${{ github.triggering_actor }}-${{ github.run_id }}
+          aws-region: us-west-2
+
+      - name: Deploy
+        if: ${{ github.event_name == 'pull_request' }}
+        id: build-site
+        env:
+          NODE_OPTIONS: "--max_old_space_size=6144"
+          AWS_REGION: us-west-2
+        run: |
+          set -eux
+          HEAD_SHA=${{ github.event.pull_request.head.sha }}
+          [ -z "$AWS_ACCESS_KEY_ID" ] && exit 1
+          BUCKET_PATH=$(jq -r .pull_request.number "$GITHUB_EVENT_PATH" | grep -E '^[0-9]+$' || :)
+          # [ -z "$BUCKET_PATH" ] || { export HUGO_BASEURL="https://${BUCKET_PATH}.pr.leo.bravesoftware.com/"; }
+          npm run build-storybook
+          echo "::group::Upload to AWS S3"
+          cd ./storybook-static
+          aws configure set default.s3.max_concurrent_requests 100
+          aws configure set default.s3.max_queue_size 10000
+          aws s3 sync --size-only . "s3://${S3_BUCKET_NAME}/${BUCKET_PATH:+pr-$BUCKET_PATH/}${HEAD_SHA}" --delete --exclude 'pr-*/*'
+          echo "::endgroup::"
+      - uses: actions/github-script@v6
+        if: ${{ github.event_name == 'pull_request' }}
+        env:
+          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+        with:
+          script: |
+            console.log(context)
+            await github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `👋 Thanks for Submitting! This PR is available for preview at the link below.\n\n✅ Preview: https://leo.bravesoftware.com/pr-${context.issue.number}/${context.payload.after}\n\nThe username and password (if not on VPN) can be found in the 1P Team vault under \`Brave.com Staging/Previews\`.`
```

Comment:
Since this is public, we can remove the part about credentials.

```suggestion
              body: `👋 Thanks for Submitting! This PR is available for preview at the link below.\n\n✅ Preview: https://leo.bravesoftware.com/pr-${context.issue.number}/${context.payload.after}`
```